src/gui/CMakeLists.txt

```diff
@@ -174,6 +174,9 @@ if (APPLE)
         ../3rdparty/qtmacgoodies/src/macstandardicon.mm
         ../3rdparty/qtmacgoodies/src/macwindow.mm
         )
+    # We want to access Cocoa specific structures in the code above
+    # and need the platform plugin interface for that - which is private.
+    include_directories(${Qt5Gui_PRIVATE_INCLUDE_DIRS})
```

Comment:
I would suggest to use ``target_include_directories(${APPLICATION_EXECUTABLE} ${Qt5Gui_PRIVATE_INCLUDE_DIRS})`` or even better  ``target_link_libraries(${APPLICATION_EXECUTABLE} Qt5::GuiPrivate)``instead.

Reply:
Thanks for the update! Specifying the target does not work here - cmake complains `Cannot specify link libraries for target "owncloud" which is not built by this project.` ( or `include directories` for the first suggestion by @dschmidt. I get it work by modifying existing `target_link_libraries` command:

```Diff
diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index 9eaf326f8..45de10b2e 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -272,7 +272,7 @@ set_target_properties( ${APPLICATION_EXECUTABLE} PROPERTIES
         RUNTIME_OUTPUT_DIRECTORY  ${BIN_OUTPUT_DIRECTORY}
 )

-target_link_libraries( ${APPLICATION_EXECUTABLE} Qt5::Widgets Qt5::Network Qt5::Xml)
+target_link_libraries( ${APPLICATION_EXECUTABLE} Qt5::GuiPrivate Qt5::Widgets Qt5::Network Qt5::Xml)
 target_link_libraries( ${APPLICATION_EXECUTABLE} ${synclib_NAME} )
 target_link_libraries( ${APPLICATION_EXECUTABLE} updater )
 target_link_libraries( ${APPLICATION_EXECUTABLE} ${OS_SPECIFIC_LINK_LIBRARIES} )
```
And of course, there's no crash with the patched app.